src/BuildScriptGeneratorCli/Commands/PrepareEnvironmentCommand.cs

```diff
@@ -0,0 +1,350 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator;
+using Microsoft.Oryx.BuildScriptGeneratorCli.Options;
+using Microsoft.Oryx.Common;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli
+{
+    [Command(Name, Description = "Sets up environment by detecting and installing platforms.")]
+    internal class PrepareEnvironmentCommand : CommandBase
+    {
+        public const string Name = "prep";
+        private const string SourceDirectoryTemplate = "-s|--src";
+        private const string SkipDetectionTemplate = "--skip-detection";
+        private const string PlatformsAndVersionsTemplate = "--platforms-and-versions";
+        private const string PlatformsAndVersionsFileTemplate = "--platforms-and-versions-file";
+
+        [Option(
+            SourceDirectoryTemplate,
+            CommandOptionType.SingleValue,
+            Description = "The source directory.")]
+        [DirectoryExists]
+        public string SourceDir { get; set; }
+
+        [Option(
+            SkipDetectionTemplate,
+            CommandOptionType.NoValue,
+            Description = "Skip detection of platforms and install the platforms.")]
+        public bool SkipDetection { get; set; }
+
+        [Option(
+            PlatformsAndVersionsTemplate,
+            CommandOptionType.SingleValue,
+            Description =
+            "Comma separated values of platforms and versions to be installed. " +
+            "Example: dotnet=3.1.200,php=7.4.5,node=2.3")]
+        public string PlatformsNamesAndVersions { get; set; }
+
+        [Option(
+            PlatformsAndVersionsFileTemplate,
+            CommandOptionType.SingleValue,
+            Description =
+            "A .env file which contains list of platforms and the versions that need to be installed. " +
+            "Example: \ndotnet=3.1.200\nphp=7.4.5\nnode=2.3")]
+        public string PlatformsNamesAndVersionsFile { get; set; }
+
+        internal override int Execute(IServiceProvider serviceProvider, IConsole console)
+        {
+            var logger = serviceProvider.GetRequiredService<ILogger<PrepareEnvironmentCommand>>();
+            var options = serviceProvider.GetRequiredService<IOptions<BuildScriptGeneratorOptions>>().Value;
+
+            var beginningOutputLog = GetBeginningCommandOutputLog();
+            console.WriteLine(beginningOutputLog);
+
+            int exitCode;
+            using (var timedEvent = logger.LogTimedEvent("EnvSetupCommand"))
+            {
+                var context = BuildScriptGenerator.CreateContext(serviceProvider, operationId: null);
+
+                IEnumerable<PlatformDetectorResult> detectedPlatforms = null;
+                if (SkipDetection)
+                {
+                    console.WriteLine(
+                        $"Skipping platform detection since '{SkipDetectionTemplate}' switch was used...");
```

Comment:
What does SkipDetectionTemplate do? What kind of scenarios will a user use this feature?

Reply:
For example, AppService folks wanted to install the 'popular' sdks upfront in an image to boost build performance of apps. Since they might install it upfront (ex: calling `RUN oryx prep --skip-detection --platforms-and-versions dotnet=3.1.300,python=3.7.7, node=12.16.3` in their Dockerfile), they need not have an app to do it and skip detection serves this purpose

Reply:
Got it, that sounds more like an `oryx install-sdk` command to me.

Reply:
Since it's part of preparation process so it makes sense.